cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/AuthorizeAction.java

```diff
@@ -209,9 +211,18 @@ private void doOpenIDCodeFlow() throws Exception {
         Subject s = AuthenticationUtil.getCurrentSubject();
         AuthMethod authMethod = AuthenticationUtil.getAuthMethodFromCredentials(s);
         if (authMethod.equals(AuthMethod.ANON)) {
-            
+            // Get the service URL for ac
+            RegistryClient regClient = new RegistryClient();
+            URL regCapabilitiesURL = regClient.getAccessURL(new URI("ivo://cadc.nrc.ca/reg"));
+
+            String regCapURLStr = regCapabilitiesURL.toString();
+            String oidcLoginHostURL = regCapURLStr.replace("/reg/capabilities","");
```

Comment:
Can you use regCapabilitiesURL.getHost() instead of using replace?  Depending on /reg/capabilities to be there makes it slightly less stable.  You may need to extract the 'scheme' too.

Reply:
Got that. I'd fallen into the pattern of what was done in the javascript. oops!

Not sure about extracting the scheme? The service returns URLs for the appropriate login host on testing.